src/main/java/AssetExchangeTransaction.java

```diff
@@ -0,0 +1,110 @@
+                .withValue(new TextValue("name", name))
+                .withValue(new IntValue("balance", balance))
+                .forNamespace(NAMESPACE).forTable(USER_TABLE_NAME);
+        tx.put(userPut);
+        tx.commit();
+        return userId;
+    }
+
+    public void exchangeAsset(String assetId, String fromId, String toId) throws TransactionException {
+        System.out.println(assetId);
+        System.out.println(fromId);
+        System.out.println(toId);
+        DistributedTransaction tx = txService.start();
+
+        Get assetGet = new Get(new Key(new TextValue(ID, assetId)))
+                .forNamespace(NAMESPACE).forTable(ASSET_TABLE_NAME);
+        Optional<Result> assetResult = tx.get(assetGet);
+        String assetOwnerId;
+        int assetPrice;
+        if (assetResult.isPresent()) {
+            assetOwnerId = ((TextValue) (assetResult.get().getValue("belong").get())).getString().get();
+            assetPrice = ((IntValue) (assetResult.get().getValue("price").get())).get();
+        } else throw new RuntimeException("Asset with id " + assetId + " belong to user id " + fromId + " doesn't exist!");
+        if (!assetOwnerId.equals(fromId)) throw new RuntimeException("Asset " + assetId +
+                " doesn't belong to user id " + fromId);
+
+        Get fromGet = new Get(new Key(new TextValue(ID, fromId))).forNamespace(NAMESPACE).forTable(USER_TABLE_NAME);
+        Get toGet = new Get(new Key(new TextValue(ID, toId))).forNamespace(NAMESPACE).forTable(USER_TABLE_NAME);
+        Optional<Result> fromResult = tx.get(fromGet);
+        Optional<Result> toResult = tx.get(toGet);
+        int fromBalance;
+        int toBalance;
+        if (fromResult.isPresent()) {
+            fromBalance = ((IntValue) (fromResult.get().getValue("balance").get())).get();
+        } else throw new RuntimeException("User with id " + fromId + " doesn't exist!");
+        if (toResult.isPresent()) {
+            toBalance = ((IntValue) (toResult.get().getValue("balance").get())).get();
+        } else throw new RuntimeException("User with id " + fromId + " doesn't exist!");
+        if (toBalance < assetPrice) throw new RuntimeException("Balance of user id" + toId + " is insufficient!");
+        Put fromPut = new Put(new Key(new TextValue(ID, fromId)))
+                .withValue(new IntValue("balance", fromBalance + assetPrice)).forNamespace(NAMESPACE).forTable(USER_TABLE_NAME);
+        Put toPut = new Put(new Key(new TextValue(ID, toId)))
+                .withValue(new IntValue("balance", toBalance - assetPrice)).forNamespace(NAMESPACE).forTable(USER_TABLE_NAME);
+        tx.put(fromPut);
+        tx.put(toPut);
+
+        txService.with(NAMESPACE, ASSET_TABLE_NAME);
+        Put assetPut = new Put(new Key(new TextValue(ID, assetId)))
+                .withValue(new TextValue("belong", toId))
+                .forNamespace(NAMESPACE).forTable(ASSET_TABLE_NAME);
+        tx.put(assetPut);
+
+        tx.commit();
```

Comment:
commit() throws CommitConflictException and it should be caught for the same reason.